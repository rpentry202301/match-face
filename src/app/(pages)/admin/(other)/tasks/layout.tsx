export const metadata = {
  title: {
    default: "タスク一覧",
    // Todo: 正式なアプリ名に差し替え
    template: `%s | 管理者画面 | Match Face`,
  },
  description: "管理者質問・解答の追加/編集ページ",
};

export default function TasksLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
