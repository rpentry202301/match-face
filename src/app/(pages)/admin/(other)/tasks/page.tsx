import TasksIndex from "@/components/pages/admin/tasks/index/TasksIndex";

/**
 * @author Hayato Kobayashi
 */
const TasksPage = ({ searchParams }: Props) => {
  return <TasksIndex searchParams={searchParams} />;
};

type Props = {
  searchParams: {
    departmentId: string | string[] | undefined;
    searchKeyword: string | string[] | undefined;
  };
};

export default TasksPage;
