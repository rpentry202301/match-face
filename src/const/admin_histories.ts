export const skills = [
    {id:1,name:"JavaScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,name:"TypeScript",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,name:"React",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,name:"Next.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,name:"Vue.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,name:"Angular",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:7,name:"Nuxt.js",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:8,name:"jQuery",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:9,name:"Redux",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:10,name:"vuex",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:11,name:"MaterialUI",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:12,name:"tailwind",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:13,name:"Bootstrap",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:14,name:"Vercel",created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'}
]

export const departments = [
    {id:1,name:"FR",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:2,name:"Java",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:3,name:"PHP",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:4,name:"CL",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:5,name:"ML",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
    {id:6,name:"QA",created_user:"rakus",created_at:"2023-07-10T16:52:46.053Z",update_user:'rakus',update_at:"2023-07-10T16:52:46.053Z"},
]

export const answer_request_groups =[
    {id:1,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:2,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:3,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:4,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:5,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
    {id:6,created_user:'rakus',created_at:'2023-07-10T16:52:46.053Z',update_user:'rakus',update_at:'2023-07-10T16:52:46.053Z'},
]
