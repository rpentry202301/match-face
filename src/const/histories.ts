export type DataType = {
  id: number;
  comment_created_at: boolean;
  name: string;
  detail: string;
  answer_update_at: string;
  skill_id: number[];
}[];

export type SkillType = {
  id: number;
  name: string;
  createdUser: string;
  createdAt: string;
  updateUser: string;
  updateAt: string;
}[];

export const projects: DataType = [
  {
    id: 1,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 2,
    comment_created_at: false,
    name: "日暮らし、",
    detail:
      "硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [2, 3, 4],
  },
  {
    id: 3,
    comment_created_at: false,
    name: "親譲りの無鉄砲で",
    detail:
      "小供の時から損ばかりしている。小学校に居る時分学校の二階から飛び降りて一週間ほど腰を抜かした事がある。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [5, 6, 7],
  },
  {
    id: 4,
    comment_created_at: true,
    name: "なぜそんな無闇を",
    detail:
      "したと聞く人があるかも知れぬ。別段深い理由でもない。新築の二階から首を出していたら、同級生の一人が冗談に、いくら威張っても、そこから飛び降りる事は出来まい。",
    answer_update_at: "2023-07-10T16:52:46.053Z",
    skill_id: [1, 2, 6, 7],
  },
  {
    id: 5,
    comment_created_at: true,
    name: "ポムポムプリン",
    detail:
      "日本のサンリオによるゴールデン・レトリバーの男のコをモチーフにしたキャラクター。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [3, 4, 5],
  },
  {
    id: 6,
    comment_created_at: false,
    name: "弱虫やーい。",
    detail:
      "と囃したからである。小使に負ぶさって帰って来た時、おやじが大きな眼をして二階ぐらいから飛び降りて腰を抜かす奴があるかと云ったから",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
  {
    id: 7,
    comment_created_at: true,
    name: "つれづれなるまゝ",
    detail:
      "日暮らし、硯にむかひて、心にうつりゆくよしなし事を、そこはかとなく書きつくれば、あやしうこそものぐるほしけれ。",
    answer_update_at: "2023-10-10T16:52:46.053Z",
    skill_id: [1, 2, 3],
  },
];

// export {data};

export const skills = [
  {
    id: 1,
    skill: "JavaScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 2,
    skill: "TypeScript",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 3,
    skill: "React",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 4,
    skill: "Next.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 5,
    skill: "Vue.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 6,
    skill: "Angular",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 7,
    skill: "Nuxt.js",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 8,
    skill: "jQuery",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 9,
    skill: "Redux",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 10,
    skill: "vuex",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 11,
    skill: "MaterialUI",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 12,
    skill: "tailwind",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 13,
    skill: "Bootstrap",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
  {
    id: 14,
    skill: "Vercel",
    created_user: "rakus",
    created_at: "2023-07-10T16:52:46.053Z",
    update_user: "rakus",
    update_at: "2023-07-10T16:52:46.053Z",
  },
];
