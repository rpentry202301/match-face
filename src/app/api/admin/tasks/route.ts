import { Answer_RequestsType } from "@/types/admin/tasks/types";
import { NextResponse } from "next/server";

/**
 * @author Hayato Kobayashi
 * @param req Next API request
 * @returns Next API responce
 *
 * @todo_1 検索ワード(searchKeyword), 所属(departmentId)で検索条件を指定
 */
export async function GET(req: Request) {
  // todo_1
  const res = await fetch(`${process.env.BE_URL}/answer_requests`, {
    method: "GET",
    headers: {
      "Content-Type": "application/json",
    },
  });
  // console.log("res", res.status);
  const answerRequests: Answer_RequestsType = await res.json();
  const tasks = answerRequests.answerRequests; // fetch時に"answerRequests.answerRequestsの形になるので修正"
  return NextResponse.json(tasks);
}
