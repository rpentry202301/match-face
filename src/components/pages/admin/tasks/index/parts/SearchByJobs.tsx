"use client";
import Input from "@/components/ui/Input";
import OrangeButton from "@/components/ui/button/OrangeButton";
import WhiteCheckButton from "@/components/ui/button/WhiteCheckButton";
import { SyntheticEvent, useMemo, useState } from "react";
import { useRouter } from "next/navigation";
import { Departments } from "@/types/admin/tasks/types";

/**
 * @author Hayato Kobayashi
 * @todo_1 レスポンシブ対応
 */
const SearchByJobs = ({ departments }: Props) => {
  const [jobsFilter, setJobsFilter] = useState<number[]>([]); // 職種フィルター状態管理
  const [inputVal, setInputVal] = useState<string>(""); // 検索入力値状態管理

  const router = useRouter();

  // 検索条件に設定するクエリ
  const query = useMemo(() => {
    if (jobsFilter.length === 0 && inputVal.length === 0) return "";

    /**
     * ワード検索クエリ
     * ex) "a b c" → "searchKeyword=a&searchKeyword=b&searchKeyword=c"
     */
    const searchKeyword =
      inputVal.length !== 0
        ? inputVal // ex)"a b c" → "a_b_c"
            .split(/[\s]+/)
            .filter((word) => word.length !== 0)
            .map((word, i) => {
              if (i === 0) return `searchKeyword=${word}`;
              return `&searchKeyword=${word}`;
            })
            .join("")
        : "";
    /**
     * 職種フィルター検索クエリ
     * ex) [1, 2, 3] → "departmentId=1&departmentId=2&departmentId=3"
     * @note searchKeywordがある場合は先頭に"&"がつく
     */
    const departmentId =
      jobsFilter.length !== 0
        ? jobsFilter
            .map((job, i) => {
              if (i === 0) return `${searchKeyword && "&"}departmentId=${job}`;
              return `&departmentId=${job}`;
            })
            .join("")
        : "";

    const searchQuery = searchKeyword + departmentId;
    return searchQuery;
  }, [jobsFilter, inputVal]);

  // 検索入力値を適用
  const handleSearch = (e: SyntheticEvent) => {
    e.preventDefault();
    router.push(`/admin/tasks${query && "?" + query}`);
  };

  const handleSetFilter = (departmentId: number) => {
    // 職種レコードを挿入
    if (!jobsFilter.includes(departmentId)) {
      const newArr = [...jobsFilter];
      newArr.push(departmentId);
      setJobsFilter(newArr);
    } else {
      // 職種レコードを削除
      const newArr = jobsFilter.filter((job) => {
        return job !== departmentId;
      });
      setJobsFilter(newArr);
    }
  };

  return (
    <div className="border-2 rounded-md w-3/6 mx-auto mt-6 p-3">
      {/* 職種フィルター入力欄 */}
      <div className="w-full mb-4">
        <div className="mb-2">
          <h2 className="text-base ml-3">▶️募集職種で検索</h2>
        </div>
        <div className="flex justify-center">
          {departments.map((department) => {
            return (
              <WhiteCheckButton
                key={department.id}
                label={department.name}
                className="text-xs w-16 mx-3"
                onClick={() => handleSetFilter(department.id)}
                data-testid={`btn_${department.name}`}
              />
            );
          })}
        </div>
      </div>

      {/* 案件名/ユーザー名検索欄 */}
      <div className="w-full">
        <div className="mb-2">
          <h2 className="text-base ml-3">▶️案件名/ユーザー名で検索</h2>
        </div>
        <form onSubmit={handleSearch} className="flex flex-col items-center">
          <Input
            id="search"
            className="border-light-gray text-xs p-1 w-[35rem]"
            onChange={(e) => setInputVal(e.target.value)}
            data-testid="search-box"
          />
          <OrangeButton
            label="検索"
            className="w-20 text-sm mt-2"
            type="submit"
          />
        </form>
      </div>
    </div>
  );
};

type Props = {
  departments: Departments[];
};

export default SearchByJobs;
