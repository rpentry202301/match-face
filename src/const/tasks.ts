import { Answer_RequestsType } from "@/types/admin/tasks/types";

// テスト用データ
export const departments = [
  {
    id: 1,
    name: "Java",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 2,
    name: "PHP",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 3,
    name: "FR",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 4,
    name: "CL",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 5,
    name: "ML",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
  {
    id: 6,
    name: "QA",
    createtdUser: "hoge",
    createdAt: new Date(),
    updateUser: "hoge",
    updateAt: new Date(),
  },
];

export const answer_requests: Answer_RequestsType = {
  answerRequests: [
    {
      id: 1,
      project: {
        id: 1,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 1,
        name: "Java",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 1,
          userId: 1,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 2,
      project: {
        id: 2,
        name: "バックエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 2,
        name: "PHP",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 2,
          userId: 2,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
    {
      id: 3,
      project: {
        id: 3,
        name: "フロントエンド案件",
        detail: "販促アプリの新規開発、既存システムの保守・運用。",
        enterpriseId: 0,
        departmentId: 0,
        questionList: [],
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
        deleted: false,
      },
      department: {
        id: 3,
        name: "FR",
        createdUser: "",
        createdAt: "-999999999-01-01T00:00:00",
        updateUser: "",
        updateAt: "-999999999-01-01T00:00:00",
      },
      answerUserList: [
        {
          answerRequestId: 3,
          userId: 3,
          userName: "テスト太郎",
          answered: false,
        },
      ],
      questionCount: 2,
      deadline: "2023-12-01T18:00:00",
      createdUser: "テスト用",
      createdAt: "2023-09-11T11:51:22.182407",
      updateUser: "テスト用",
      updateAt: "2023-09-11T11:51:22.182407",
    },
  ],
};
