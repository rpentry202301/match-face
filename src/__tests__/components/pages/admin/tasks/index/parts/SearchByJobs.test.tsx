import SearchByJobs from "@/components/pages/admin/tasks/index/parts/SearchByJobs";
import { departments } from "@/const/tasks";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import { useRouter } from "next/navigation";
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: jest.fn(),
  }),
}));

describe("TaskList.tsx", () => {
  beforeAll(() => {
    jest.clearAllMocks();
    cleanup();
  })
  afterAll(() => {
    jest.clearAllMocks();
    cleanup();
  });

  it("スナップショットテスト", () => {
    const { container } = render(<SearchByJobs departments={departments} />);
    expect(container).toMatchSnapshot();
  });

  describe("入力テスト", () => {
    beforeEach(() => {
      cleanup();
      render(<SearchByJobs departments={departments} />);
    });
    it("検索ボックス入力テスト", () => {
      const input = screen.getByTestId("search-box") as HTMLInputElement;
      fireEvent.change(input, { target: { value: "hoge foo" } });
      // 正常に入力されているかテスト
      expect(input.value).toBe("hoge foo");
    });
    it("職種フィルターボタンテスト", () => {
      const phpBtn = screen.getByTestId("btn_PHP");
      fireEvent.click(phpBtn); // "PHP"ボタン押下
      const frBtn = screen.getByTestId("btn_FR");
      fireEvent.click(frBtn); // "FR"ボタン押下

      // ボタン押下時のクラス
      const clickedClass =
        "bg-gray-200 translate-y-0.5 shadow-sm py-2 px-4 rounded-full border";

      expect(phpBtn).toHaveClass(clickedClass);
      expect(frBtn).toHaveClass(clickedClass);

      const javaBtn = screen.getByTestId("btn_Java");
      const clBtn = screen.getByTestId("btn_CL");
      const qaBtn = screen.getByTestId("btn_QA");
      const mlBtn = screen.getByTestId("btn_ML");
      expect(javaBtn).not.toHaveClass(clickedClass);
      expect(clBtn).not.toHaveClass(clickedClass);
      expect(qaBtn).not.toHaveClass(clickedClass);
      expect(mlBtn).not.toHaveClass(clickedClass);
    });

    // "検索"ボタン押下時のrouter.push()の挙動をテストしたかったが断念。
    
  });
});
