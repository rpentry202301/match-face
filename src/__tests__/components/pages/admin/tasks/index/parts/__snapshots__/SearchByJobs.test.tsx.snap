// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskList.tsx スナップショットテスト 1`] = `
<div>
  <div
    class="border-2 rounded-md w-3/6 mx-auto mt-6 p-3"
  >
    <div
      class="w-full mb-4"
    >
      <div
        class="mb-2"
      >
        <h2
          class="text-base ml-3"
        >
          ▶️募集職種で検索
        </h2>
      </div>
      <div
        class="flex justify-center"
      >
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_Java"
        >
          Java
        </button>
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_PHP"
        >
          PHP
        </button>
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_FR"
        >
          FR
        </button>
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_CL"
        >
          CL
        </button>
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_ML"
        >
          ML
        </button>
        <button
          class="bg-white hover:bg-gray-100 py-2 px-4 rounded-full border shadow-md text-xs w-16 mx-3"
          data-testid="btn_QA"
        >
          QA
        </button>
      </div>
    </div>
    <div
      class="w-full"
    >
      <div
        class="mb-2"
      >
        <h2
          class="text-base ml-3"
        >
          ▶️案件名/ユーザー名で検索
        </h2>
      </div>
      <form
        class="flex flex-col items-center"
      >
        <div>
          <input
            class="border border-light-gray text-xs p-1 w-[35rem]"
            data-testid="search-box"
            id="search"
          />
        </div>
        <div>
          <button
            class="bg-orange drop-shadow-md hover:saturate-150 active:drop-shadow-none active:shadow-inner hover:bg-depp-orange active:bg-depp-orange active:translate-y-0.5 text-white rounded-xl p-2 h-auto w-20 text-sm mt-2"
            type="submit"
          >
            検索
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
